cmd/hub/cli/version.go

```diff
@@ -53,7 +52,9 @@ var versionCmd = &cobra.Command{
 	Long:  `Shows the installed CLI version.`,
 	Args:  cobra.ExactArgs(0),
 	Run: func(c *cobra.Command, args []string) {
-		version := bi.Version
+		version := bi.GitSummary
```

Comment:
Let me know if this is what you had in mind @andrewxhill and @sanderpick

Reply:
i think going in circles a bit. probably better to start from the other side and answer these three and then the solution is as simple as it looks.

1. What's the expected behavior if someone downloads from the release page, installs, and calls `version`?
2. What's the expected behavior if someone `make build` and calls `version`?
3. What's the expected behavior if someone `go install ./...` and calls `version`?

Reply:
sorry, `behavior => result` in those three questions

Reply:
Yep, that's what I was thinking @asutula. 

@andrewxhill, here's what I think:
1. Should be the result of `GitSummary` (govvv used in CI)
1. Should be the result of `GitSummary` (govvv used in local Makefile)
1. Should be the default value of `GitSummary`, perhaps "unknown"

Reply:
k, for #3 that's what i was getting at in my comment over the weekend. current default GitSummary is "" so needs an update.

i think #1 or #2 could be broken here

Reply:
If we want 1 and 2, that is the way it works in this current revision. For 3, it will currently print `""`, then the semver parsing will fail and the resulting `Running a custom hub build...` will be displayed. So I guess the only thing to change there is to print something better than `""` for 3? We can just change the default in `buildinfo`. I actually don't mind the idea of making all the defaults in there `"git"` instead of `""`.

Reply:
@andrewxhill I don't think 1 and 2 are broken, so if you see how they are, please explain.

Reply:
kk, is there something new to setup maybe?

<img width="1002" alt="Screen Shot 2020-09-21 at 9 56 39 AM" src="https://user-images.githubusercontent.com/370259/93797185-bffd6900-fbf0-11ea-93a1-be4d2fd932d6.png">

Reply:
Another idea to float... In powergate, we have the equivalent of the `version` command just print out all the information from `buildinfo`. Looks like:
```
pow --version
pow build info:
        version:        git
        build date:     2020-09-21T15:01:22Z
        git summary:    v0.6.0-dirty
        git branch:     master
        git commit:     a928c57
        git state:      dirty
```
This seems like a good option to me because we don't have to make any decisions about what buildinfo value to assign to `version`. It provides a lot more context. In this case above, I can infer that the user ran a local `make` build, has some local changes on top of the `v0.6.0` tag, and I have the commit hash and all.

Reply:
Oh I see. Yea, the `-dirty` in GitSummary is breaking the semantic version parser I guess. Yes, that is broken indeed. See my above comment. I think this could be a solution.

Reply:
Also, I'm a little confused about what `make build` actually does. It runs `go build ./...` but I don't think that installs the binaries in you PATH. It looks like you're assuming it does. Might want to use `make install` instead just to be sure you're installing the binaries.

Reply:
oh nice, read that wrong, thought it had changed. better!

Reply:
the semantic version parsing was really helpful so you could easily check if the user is behind production. which is only useful if it's not dirty. so if it fails you could assume it's a custom build and skip the remote check perhaps... though it still seems valuable if someone is on 0.6.0-dirty and production is on 0.7.0 to give them the warning

Reply:
(or at least print remote still)